File: mobile/lib/features/settings/data/settings_model.dart
class SettingsModel {
  final String shopName;
  final String currency;
  final double taxRate;
  final String receiptFormat;
  final String address;
  final String phone;

  SettingsModel({
    required this.shopName,
    required this.currency,
    required this.taxRate,
    required this.receiptFormat,
    required this.address,
    required this.phone,
  });

  factory SettingsModel.fromJson(Map<String, dynamic> json) {
    return SettingsModel(
      shopName: json['shopName'] ?? 'Smart Dress Shop',
      currency: json['currency'] ?? 'INR',
      taxRate: (json['taxRate'] as num?)?.toDouble() ?? 0.0,
      receiptFormat: json['receiptFormat'] ?? 'THERMAL',
      address: json['address'] ?? '',
      phone: json['phone'] ?? '',
    );
  }
}

File: server/src/modules/settings/settings.model.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IShopSettings extends Document {
  shopName: string;
  currency: string;
  taxRate: number;
  receiptFormat: 'STANDARD' | 'THERMAL' | 'A4';
  address: string;
  phone: string;
}

const shopSettingsSchema = new Schema<IShopSettings>(
  {
    shopName: { type: String, required: true, default: 'Smart Dress Shop' },
    currency: { type: String, required: true, default: 'INR' },
    taxRate: { type: Number, required: true, default: 0 },
    receiptFormat: { type: String, enum: ['STANDARD', 'THERMAL', 'A4'], default: 'THERMAL' },
    address: { type: String, default: '' },
    phone: { type: String, default: '' },
  },
  { timestamps: true }
);

export const ShopSettings = mongoose.model<IShopSettings>('ShopSettings', shopSettingsSchema);
